CodeGeneration/Sources/SyntaxSupport/AvailabilityNodes.swift

```diff
@@ -118,42 +118,54 @@ public let AVAILABILITY_NODES: [Node] = [
     ]
   ),
 
-  // version-tuple -> integer-literal
-  //                | float-literal
-  //                | float-literal '.' integer-literal
+  // version-tuple-element -> '.' float-literal |
+  //                          '.' interger-literal
+
   Node(
-    name: "VersionTuple",
-    nameForDiagnostics: "version tuple",
-    description: "A version number of the form major.minor.patch in which the minor and patch part may be omitted.",
+    name: "VersionTupleElement",
```

Comment:
What do you think about naming this `VersionComponent` and the list `VersionComponentList`? And then rename `subversions` in your implementation to `components`. Subversion still always reminds me of the version control system.

Otherwise, I think this is a great improvement.

Reply:
I think these names are better. Already renamed these nodes.